<a href="https://colab.research.google.com/github/Imran1hp/Adhaar-Data-Analysis/blob/main/Adhaar_dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import requests
import matplotlib.pyplot as plt
import seaborn
from pathlib import Path

# Download the data

In [ ]:
from google.colab import drive
drive.mount('/content/drive')


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
dataset_dir="//content/drive/MyDrive/dataset for adhaar-20260108T055715Z-1-001/dataset for adhaar"
dataset_dir

'//content/drive/MyDrive/dataset for adhaar-20260108T055715Z-1-001/dataset for adhaar'

In [ ]:
dataset_path = Path(dataset_dir)

In [ ]:
import os
os.listdir(dataset_path)

['demo_cleaned.csv', 'bio_cleaned.csv', 'enroll_cleaned.csv']

#Loading  the data

In [ ]:
demo_df = pd.read_csv(dataset_path / 'demo_cleaned.csv')
bio_df = pd.read_csv(dataset_path / 'bio_cleaned.csv')
enroll_df = pd.read_csv(dataset_path /  'enroll_cleaned.csv')

In [ ]:
demo_df.describe().transpose()

count           mean  ...       75%       max
pincode        1598012.0  526610.206336  ...  691572.0  855456.0
demo_age_5_17  1598012.0       2.251381  ...       2.0    2690.0
demo_age_17_   1598012.0      20.650443  ...      15.0   16166.0

[3 rows x 8 columns]

In [ ]:
bio_df.describe().transpose()

count           mean  ...       75%       max
pincode       1766159.0  521724.856948  ...  686673.0  855456.0
bio_age_5_17  1766159.0      18.943164  ...      11.0    8002.0
bio_age_17_   1766159.0      19.706246  ...      10.0    7625.0

[3 rows x 8 columns]

In [ ]:
enroll_df.describe().transpose()

count           mean  ...       75%       max
pincode         983000.0  518455.375778  ...  700092.0  855456.0
age_0_5         983000.0       3.534392  ...       3.0    2688.0
age_5_17        983000.0       1.720134  ...       1.0    1812.0
age_18_greater  983000.0       0.169341  ...       0.0     855.0

[4 rows x 8 columns]

In [ ]:
enroll_df

date          state  ... age_5_17  age_18_greater
0       2025-03-02      meghalaya  ...       61              37
1       2025-03-09      karnataka  ...       33              39
2       2025-03-09  uttar pradesh  ...       82              12
3       2025-03-09  uttar pradesh  ...       29              15
4       2025-03-09      karnataka  ...       16              21
...            ...            ...  ...      ...             ...
982995  2025-12-31    west bengal  ...        0               0
982996  2025-12-31    west bengal  ...        2               0
982997  2025-12-31    west bengal  ...        1               0
982998  2025-12-31    west bengal  ...        0               0
982999  2025-12-31    west bengal  ...        1               0

[983000 rows x 7 columns]

In [ ]:
demo_state_names = demo_df["state"].unique().tolist()
bio_state_names = bio_df["state"].unique().tolist()
enroll_state_names = enroll_df["state"].unique().tolist()


In [ ]:
len(demo_state_names),len(bio_state_names),len(enroll_state_names)

(58, 50, 49)

There are 58 unique state in demo_df , 50 unique state name is bio_state ,49 unique state name is eroll_state names.


In [ ]:
demo_dist_names = demo_df["district"].unique().tolist()
bio_dist_names = bio_df["district"].unique().tolist()
enroll_dist_names = enroll_df["district"].unique().tolist()

In [ ]:
len(demo_dist_names), len(bio_dist_names),len(enroll_dist_names)

(961, 949, 964)

There are 961 district  in demo_df , there are 949 district in bio_df and 964 district in enroll_df

In [ ]:
demo_df.date.min() , demo_df.date.max()

('2025-03-01', '2025-12-29')

In [ ]:
bio_df.date.min(), bio_df.date.max()

('2025-03-01', '2025-12-29')

In [ ]:
enroll_df.date.min(),enroll_df.date.max()

('2025-03-02', '2025-12-31')

Demo_df and bio_df start from "2025-03-01" end at "2025-12-29" and enroll_df start from "2025-03-02" end at "2025-12-31"